luna/featurevis/transformations.py

```diff
@@ -3,53 +3,118 @@
+            )
         else:
-            img_noise = tf.random.uniform((img.shape),
-                                          dtype=tf.dtypes.float32)
+            img_noise = tf.random.uniform((img.shape), dtype=tf.dtypes.float32)
         img_noise = (img_noise - 0.5) * 0.25 * random.random()
         img = img + img_noise
         img = tf.clip_by_value(img, -1, 1)
     return img
 
 
-def blur_image(img, blur):
+def pad(img, pad_size, pad_mode="REFLECT", constant_value=0.5):
+    """Pad the image
+
+    Args:
+        img (list): the image data to be padded.
+        pad_size (int): size of pad.
+        mode (str, optional): mode for padding; can be "CONSTANT", "REFLECT" or "SYMMETRIC".
+        constant_value (float, optional): The scalar pad value. Defaults to 0.5.
+
+    Returns:
+        list: the padded image.
+    """
+    img = tf.pad(
+        img,
+        [(0, 0), (pad_size, pad_size), (pad_size, pad_size), (0, 0)],
+        pad_mode,
+        constant_value,
+    )
+    return img
+
+
+def jitter(img, jitter_dist, seed=None):
+    """Jitter the image & add an artistic effect to the image.
+
+    Args:
+        img (list): the image data to be jittered.
+        jitter_dist (int): the distance of effective neighborhoods.
+        seed (int, optional): Defaults to None.
+
+    Returns:
+        list: the jittered image.
+    """
+    img = tf.convert_to_tensor(img)
+    img_shape = tf.shape(img)
+    crop_shape = tf.concat(
+        [img_shape[:-3], img_shape[-3:-1] - jitter_dist, img_shape[-1:]], 0
+    )
+    crop = tf.image.random_crop(img, crop_shape, seed=seed)
+    img_shape_list = img.get_shape().as_list()
+    mid_shp_changed = [
+        img_shape_list[-3] - jitter_dist if img_shape_list[-3] is not None else None,
+        img_shape_list[-2] - jitter_dist if img_shape_list[-3] is not None else None,
+    ]
+    crop.set_shape(img_shape_list[:-3] + mid_shp_changed + img_shape_list[-1:])
+
+    return crop
+
+
+def blur(img, add_blur):
     """Gaussian blur the image to be modified.
 
     Args:
         img (list): the image to be blurred.
-        blur (boolean): whether to blur the image.
+        add_blur (boolean): whether to blur the image.
     Returns:
         list: the blurred image.
     """
-    if blur:
+    if add_blur:
         img = gaussian_blur(img, sigma=0.001 + random.random())
         img = tf.clip_by_value(img, -1, 1)
     return img
 
 
-def rescale_image(img, scale):
+def random_select(rand_values, seed=None):
+    """Generate the upper bound on the range of random value.
+
+    Args:
+        rand_scale (list): random selection for upper bound
+        seed (int, optional): for reproducibility. Defaults to None.
+
+    Returns:
+        list: list of random values
+    """
+    scale_list = list(rand_values)
+    rand_n = tf.random.uniform((), 0, len(scale_list), "int32", seed)
+    return tf.constant(scale_list)[rand_n]
+
+
+def rescale(img, scale):
```

Comment:
Again, could we get rid of the second param?

Reply:
Is rescale even the correct term here? Isn't it rather something like `scale_values`

Reply:
Resolved by giving transformations as function.